ophyd/areadetector/base.py

```diff
@@ -240,6 +240,34 @@ def get_asyn_digraph(self):
 
         return G, port_map
 
+    def visualize_asyn_digraph(self, *args, **kwargs):
+        '''This generates a figure showing the current asyn port layout.
+
+        This method generates a plot showing all of the currently enabled
+        Areadetector plugin asyn ports and their relationships. The current
+        ports and relationships are found using self.get_asyn_digraph.
+
+        Parameters
+        ----------
+        *args, **kwargs : networkx.draw_networkx args and kwargs.
+            For the allowed args and kwargs see the `networkx.draw_networkx documentation <https://networkx.github.io/documentation/networkx-1.10/reference/generated/networkx.drawing.nx_pylab.draw_networkx.html>`_
+        '''
+        # Importing matplotlib.pyplot here as it is not a dependency except for
+        # this method.
+        import matplotlib.pyplot as plt
+
+        # Generate the port_map Digraph.
+        G, port_map = self.get_asyn_digraph()
+        # Create and label the figure.
+        plt.figure('AD port map for {}'.format(self.name))
```

Comment:
```python
fig, ax = plt.subpplots(num='...')
```
and then passing the `ax` into the `draw_networkx` call is a bit better. 

Having `ax=None` in the signature and then only doing this logic if the user did not pass an axes would also be a good idea.

Reply:
It does not rely on the global state in pyplot to make sure the graph goes where you want it to.